[Assignment](#Assignment)

[Data Exploration](#Data-Exploration)

# Assignment

The task is to predict the day price direction of Amazon.com, Inc. (AMZN).

The stock market is very complex and highly volatile. In order to be profitable, we do not need to predict the correct price, but rather, the price direction: whether it will be higher or lower than the price that is today. If we predict it to be higher, we might as well buy some stocks, else, we should probably sell. Therefore, the target would be a binary classification whether the next day closing price will be higher than the opening price.

We have data for the period from 1997 up to the year 2020 that we have split into training (1997-2016), validation (2016-2018) and testing (2018-2020) periods. The data is available in the AMZN_train.csv, AMZN_val.csv and AMZN_test.csv files, respectively.

# Data Exploration

In our initial exploration, we will load the data sets and see what data attributes are available to us. We will also plot the variables, to see if we can find some trends in the data, and explore the possibility of engineering some additional features. We will do the data loading and analysis in pandas, so let us load that library and begin exploring.

In [4]:
import pandas as pd

In [5]:
# load the training set
df_train = pd.read_csv(r"D:\Main\My Practice and experiments\My Practice LAB\Python\2024\Statquest\Prediction of stock price direction\datasets\datasets\AMZN_train.csv")

In [6]:
# (#rows, #columns)
df_train.shape

(4781, 7)

Pandas' info() method is really useful to get a quick peek at the column names, types, non-null counts, as well as the data set's memory size! We have mostly numerical columns (5 floats, 1 int) and an object column - the date. No null values at all, and just 261 KB in size.

In [7]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4781 entries, 0 to 4780
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       4781 non-null   object 
 1   Open       4781 non-null   float64
 2   High       4781 non-null   float64
 3   Low        4781 non-null   float64
 4   Close      4781 non-null   float64
 5   Adj Close  4781 non-null   float64
 6   Volume     4781 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 261.6+ KB


The training data set consists of 4781 rows and 7 columns. These are presented above. In the cell below, we use the describe() method to get a quick overview of our data set.

In [8]:
df_train.describe()

Open         High          Low        Close    Adj Close  \
count  4781.000000  4781.000000  4781.000000  4781.000000  4781.000000   
mean    127.619845   129.480122   125.697925   127.662449   127.662449   
std     145.693083   147.132537   144.053633   145.677581   145.677581   
min       1.406250     1.447917     1.312500     1.395833     1.395833   
25%      34.299999    34.849998    33.660000    34.310001    34.310001   
50%      62.880001    64.750000    60.937500    62.750000    62.750000   
75%     189.009995   191.600006   186.210007   189.029999   189.029999   
max     717.380005   722.450012   711.510010   717.929993   717.929993   

             Volume  
count  4.781000e+03  
mean   8.225935e+06  
std    7.810188e+06  
min    4.872000e+05  
25%    4.200900e+06  
50%    6.200100e+06  
75%    9.239900e+06  
max    1.043292e+08